### ISSUE TITLE
Feature request: save log button?

### ISSUE THREAD

**Issue by kieranjol** [APTrust/dart#554]
Would it be possible to add a 'Save Log' button at the end of a job, for example bag validation and bag creation?
I'm aware of the log file but it could be useful to allow a user to save the outcome of the job, even if it's just the Job/ProcessID/Started At/Validation/outcome etc.

**Comment by diamondap** [comment]
This feature is already in the pipeline for DART 3, in which I plan to automatically save artifacts for each job. Artifacts include:

1. A summary of the outcome of the job's bagging, validation and upload steps.
2. The bag's tag files.
3. The bag's payload manifests.

I've gotten various requests for all of these over the years, so it seems to be a pretty common need. DART 3's improved storage layer makes these features much easier to implement. These artifacts will be searchable and discoverable through the UI.

**Comment by diamondap** [comment]
Update: This feature won't make it into the existing DART, but it is now fully implemented in DART 3, which still has a ways to go before release. Here's what the jobs list looks like in DART3. You'll notice an Artifacts link for each job.

![DART3-Jobs-List](https://github.com/APTrust/dart/assets/225514/c1697031-5c1f-419f-a657-092eff40b283)

If you click that, you'll see artifacts from each time this job was run listed in the left nav. Artifacts include manifests, tag files, and a JSON file describing what happened when the job ran. Click any file to view its contents. This shows the md5 manifest from the last run.

![DART3-Job-Artifacts](https://github.com/APTrust/dart/assets/225514/ee652427-3556-4c99-8f21-35a90265265f)

**Comment by kieranjol** [comment]
This is wonderful! I’m teaching a class on file transfer and fixity in University College Dublin on Thursday and I’ll show off these screenshots :)
Will there be a “save as” type button where these artefacts can be saved in a file that an archivist could store as evidence that the workflow and validation was carried out?

**Comment by diamondap** [comment]
Yes. At the bottom of the artifact page is a Copy to Clipboard button so you can copy the file contents into a text file or wherever else you want. It was cut off in the other screenshot, but you can see it here.

![DART3-Copy-Artifact](https://github.com/APTrust/dart/assets/225514/a6b289d4-0dae-471c-830c-0349a5d6e106)

**Comment by diamondap** [comment]
Would you want that button for each type of artifact? So you could save a tag file, a manifest, the job result, etc?

**Comment by kieranjol** [comment]
Personally only the job result would be useful for us as we would have the
rest of the data in the actual bag.

**Comment by diamondap** [comment]
Good to know. Thanks.

**Comment by diamondap** [comment]
Quick update. We're going to hold off on the "save logs" button for now, as we're in a period of transition with DART 3 moving to the Wails desktop framework. Here's why we're holding off on the save logs feature.

If we send the file back from the server with content-disposition=”attachment,” the browser will automatically save it to the Downloads folder, which is not what most people will want.

If we try to use the browser’s showSaveFilePicker, it won’t be supported in some browsers (notably Safari and Firefox). See https://developer.mozilla.org/en-US/docs/Web/API/Window/showSaveFilePicker

If we use Wails' [SaveFileDialog](https://wails.io/docs/reference/runtime/dialog/#savefiledialog), then we will have to provide a different implementation for the web-based version of DART. Also note that many parts of the Wails API will be changing in Wails 3.

Too many things are up in the air on this one, so we'll move it to the back burner and focus on issues we know we can fix.
